File: testes_e_Estudos/EstudosProvaPratica_3/C/HalldosAssassinos.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

typedef struct Assasino {
    char nome[50];
    int assasinados;
    bool morreu;
} Assasino;

// Função para imprimir o resultado final
void printFinal(Assasino assasino[], int n) {
    printf("HALL OF MURDERERS\n");

    // Ordenar os assassinos em ordem lexicográfica
    for (int i = 0; i < n - 1; i++) {
        for (int j = i + 1; j < n; j++) {
            if (strcmp(assasino[i].nome, assasino[j].nome) > 0) {
                Assasino temp = assasino[i];
                assasino[i] = assasino[j];
                assasino[j] = temp;
            }
        }
    }

    // Imprimir apenas os assassinos vivos
    for (int i = 0; i < n; i++) {
        if (!assasino[i].morreu) {
            printf("%s %d\n", assasino[i].nome, assasino[i].assasinados);
        }
    }
}

int main() {
    Assasino assasino[50]; // Garantir espaço suficiente
    int n = 0;
    int m = 0;

    char nome[50], mortos[50][50];

    while (scanf(" %s %s", nome, mortos[m++]) == 2) {

        // Verificar se o assassino já está na lista
        int achou = -1;
        for (int i = 0; i < n; i++) {
            if (strcmp(assasino[i].nome, nome) == 0) {
                achou = i;
            }
        }

        if (achou == -1) { // Novo assassino
            strcpy(assasino[n].nome, nome);
            assasino[n].assasinados = 1;
            assasino[n].morreu = false;
            n++;
        } else { // Assassino já está na lista
            assasino[achou].assasinados++;
        }

        // Marcar o assassinado como mortos, se ele estiver na lista
        for (int i = 0; i < n; i++) {
            for(int j = 0; j < m; j++) {
                if (strcmp(assasino[i].nome, mortos[j]) == 0) {
                    assasino[i].morreu = true;
                }
            }
        }
    }

    printFinal(assasino, n);

    return 0;
}

File: testes_e_Estudos/EstudosProvaPratica_3/Java/HalldosAssassinos.java

import java.util.*;


public class HalldosAssassinos{


    public static void printFinal(Assasino assasinos[], int n){

        System.out.println("HALL OF MURDERERS");

        for(int i = 0; i < n-1; i ++){
            for(int j = i+1; j < n; j++){
                if(assasinos[i].nome.compareTo(assasinos[j].nome) > 0){
                    Assasino tmp = assasinos[i];
                    assasinos[i] = assasinos[j];
                    assasinos[j] = tmp;
                }
            }
        }


        for(int i = 0; i < n; i++){
            if(!assasinos[i].morto){
                assasinos[i].print();
            }
        }

    }



    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        Assasino assasinos[] = new Assasino[50];
        int n = 0; int m = 0;
        String mortos[] = new String[50];

        while(sc.hasNext()){

            String nome = sc.next();
            mortos[m++] = sc.next();

            int achou = -1;

            for(int i = 0; i < n; i++){
                if(assasinos[i].nome.compareTo(nome) == 0){
                    achou = i;
                }
            }

            if(achou == -1){
                assasinos[n++] = new Assasino(nome);
            }else{
                assasinos[achou].assasinados++;
            }

            

            for(int i = 0; i < n; i++){
                for(int j = 0; j < m; j++){
                    if(assasinos[i].nome.compareTo(mortos[j]) == 0){
                        assasinos[i].morto = true;
                    }
                }
            }

        }

        printFinal(assasinos, n);
        
    }
}



class Assasino{

    public String nome;
    public int assasinados;
    public boolean morto;

    Assasino(String nome){
        this.nome = nome;
        this.assasinados = 1;
        this.morto = false;
    }

    public void print(){
        System.out.println(this.nome + " " + this.assasinados);
    }
}
